src/trie/prefix_set.rs

```diff
@@ -1,28 +1,66 @@
 use crate::trie::util::has_prefix;
-use std::collections::BTreeSet;
+use bytes::Bytes;
 
 #[derive(Clone)]
-pub(crate) struct PrefixSet(BTreeSet<Vec<u8>>);
+pub(crate) struct PrefixSet {
+    keys: Vec<Bytes>,
+    sorted: bool,
+    index: usize,
+}
 
 impl PrefixSet {
     pub(crate) fn new() -> Self {
-        Self(BTreeSet::new())
+        Self {
+            keys: Vec::new(),
+            sorted: true,
+            index: 0,
+        }
+    }
+
+    fn sort(&mut self) {
+        self.keys.sort();
+        self.keys.dedup();
```

Comment:
Missing `self.sorted = true;`?

Reply:
Yes, added it. I wonder how it could still be faster without it though.